.github/workflows/reusable_e2e_test.yml

```diff
@@ -0,0 +1,58 @@
+name: "Integration Tests"
+
+on:
+  workflow_call:
+    inputs:
+      repository:
+        required: true
+        type: string
+      branch:
+        required: true
+        type: string
+      environment:
+        required: true
+        type: string
+      os:
+        required: true
+        type: string
+
+jobs:
+  IntegrationTests:
+    if: (github.repository == 'aws-deadline/${{inputs.repository}}')
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    environment: ${{inputs.environment}}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+        if: ${{inputs.branch == 'mainline' || inputs.branch == 'release'}}
```

Comment:
Do we require this? If so, maybe we should have an equivalent branch check for the last step (Run Integration Tests)

Reply:
Not needed